app/models/iiif_content_search_response.rb

```diff
@@ -67,7 +67,7 @@ def ignored_params
   def resources
     return to_enum(:resources) unless block_given?
 
-    search.highlights.each do |id, hits|
+    search.highlights.reject { |_id, hits| hits.nil? }.each do |id, hits|
```

Comment:
Would this be a good addition to the `Search` class' `highlights` method?  I could imagine that method using `next`/`compact` if the lack of highlights in a document wasn't important (or returning an empty array in the case that the hits are nil if that is useful info).  This way consumers of `Search#highlights` won't have to concern themselves with hits being present or not.

Just a thought, either way it looks good to me.